src/gui/masterkey/YubiKeyEditWidget.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "YubiKeyEditWidget.h"
+#include "ui_YubiKeyEditWidget.h"
+#include "gui/MessageBox.h"
+#include "gui/MainWindow.h"
+#include "keys/CompositeKey.h"
+#include "keys/YkChallengeResponseKey.h"
+#include "config-keepassx.h"
+
+#include <QtConcurrent>
+
+YubiKeyEditWidget::YubiKeyEditWidget(QWidget* parent)
+    : KeyComponentWidget(parent)
+    , m_compUi(new Ui::YubiKeyEditWidget())
+{
+    setComponentName(tr("YubiKey Challenge-Response"));
+    setComponentDescription(tr("<p>If you possess a <a href=\"https://www.yubico.com/\">YubiKey</a>, you can add it "
+                               "for additional security.</p><p>The YubiKey needs to be programmed with an "
```

Comment:
Recommended wording: "If you own a YubiKey, you can use it for additional security. The YubiKey requires one of its slots to be programmed as an HMAC-SHA1 Challenge-Response."